compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -90,13 +99,59 @@ class ExtractDependencies extends Phase {
             }
         }
         extractDeps.topLevelDependencies.foreach(dep =>
-          recordDependency(sourceFile.file, dep, DependencyContext.DependencyByMemberRef))
+          recordDependency(sourceFile.file, dep._2, DependencyContext.DependencyByMemberRef)(ctx.withOwner(dep._1)))
         extractDeps.topLevelInheritanceDependencies.foreach(dep =>
-          recordDependency(sourceFile.file, dep, DependencyContext.DependencyByInheritance))
+            recordDependency(sourceFile.file, dep._2, DependencyContext.DependencyByInheritance)(ctx.withOwner(dep._1)))
+      }
+    }
+  }
+
+  private def firstClassOrModule(tree: tpd.Tree)(implicit ctx: Context): Option[Symbol] = {
+    import tpd._
+    val acc = new TreeAccumulator[Option[Symbol]] {
+      def apply(x: Option[Symbol], t: Tree)(implicit ctx: Context) =
+        if (x.isDefined) x
+        else t match {
+          case moduleDef: Thicket =>
+            Some(moduleDef.symbol)
+          case typeDef: TypeDef =>
+            Some(typeDef.symbol)
+          case other =>
+            foldOver(x, other)
+        }
+    }
+    acc(None, tree)
+  }
+
+  private def classFile(sym: Symbol)(implicit ctx: Context): Option[AbstractFile] = {
+    // package can never have a corresponding class file; this test does not
+    // catch package objects (that do not have this flag set)
+    if (sym.is(Package)) None
+    else {
+      val file = Option(sym.associatedFile)
+
+      Option(sym.associatedFile).flatMap {
```

Comment:
I don't understand what this is for, if the symbol comes from source then `sym.associatedFile` will never be `NoSourceFile`.